framework/src/messaging/messaging_sndinternal.c

```diff
@@ -0,0 +1,221 @@
+#include <debug.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/prctl.h>
+#include <sys/types.h>
+#include <messaging/messaging.h>
+#include "messaging_internal.h"
+
+#define MSG_RECV_NOT_INIT (-1)
+/****************************************************************************
+ * private functions
+ ****************************************************************************/
+static int messaging_set_async_callback(mqd_t mqdes, msg_callback_info_t *param)
+{
+	int ret;
+	msg_recv_info_t *data;
+
+	ret = messaging_set_notify_signal(SIGMSG_MESSAGING, (_sa_sigaction_t)param->cb_func);
+	if (ret != OK) {
+		return ERROR;
+	}
+
+	data = (msg_recv_info_t *)MSG_ALLOC(sizeof(msg_recv_info_t));
+	if (data == NULL) {
+		msgdbg("[Messaging] send fail : out of memory for recv info.\n");
+		return ERROR;
+	}
+	data->mqdes = mqdes;
+	data->user_cb = param->cb_func;
+	data->cb_data = param->cb_data;
+
+	ret = messaging_set_notification(SIGMSG_MESSAGING, data);
+	if (ret != OK) {
+		MSG_FREE(data);
+		return ERROR;
+	}
+	return OK;
+}
+/****************************************************************************
+ * Name : messaging_send_packet
+ * 
+ * Description:
+ *  This function calls receive function based on receive type.
+ *
+ * Input Parameters:
+ *  port_name : The message port name to send
+ *  msg       : The message to be sent
+ *  msglen    : The length of message to be sent
+ *  priority  : A non-negative integer that specifies the priority of this message
+ * 
+ * Return Value:
+ *  On success, 0 (OK) is returned.; On failure, -1 (ERROR) is returned.
+ ****************************************************************************/
+int messaging_send_packet(const char *port_name, int msg_type, msg_send_data_t *send_data, msg_callback_info_t *cb_info, int recv_cnt)
+{
+	int ret = OK;
+	mqd_t mqdes;
+	struct mq_attr internal_attr;
+	char *send_packet;
+	int send_size;
+	pid_t sender_pid;
+	int send_type;
+
+	send_size = MSG_HEADER_SIZE + send_data->msglen;
+
+	internal_attr.mq_maxmsg = CONFIG_MESSAGING_MAXMSG;
+	internal_attr.mq_msgsize = send_size;
+	internal_attr.mq_flags = 0;
+
+	mqdes = mq_open(port_name, O_WRONLY, 0666, &internal_attr);
+	if (mqdes == (mqd_t)ERROR) {
+		if (errno == ENOENT) {
+			msgdbg("[Messaging] send fail : There is no receiver.\n");
+		} else {
+			msgdbg("[Messaging] send fail : open fail %d.\n", errno);
+		}
+		return ERROR;
+	}
+
+	/* Sender can receive the reply when only one receiver waits. */
+	if (recv_cnt == 1 && msg_type == MSG_SEND_ASYNC) {
+		ret = messaging_set_async_callback(mqdes, cb_info);
+		if (ret != OK) {
+			msgdbg("[Messaging] send fail : set async callback error.\n");
+			return ERROR;
+		}
+	}
+
+	send_packet = (char *)MSG_ALLOC(send_size);
+	if (send_packet == NULL) {
+		msgdbg("[Messaging] send fail : out of memory for including header.\n");
+		return ERROR;
+	}
+
+	/* Send packet is like below.
```

Comment:
Good, I like such a style.
But, if it is possible, adding size info(byte or something)will more helpful.
message(max x kb)

Reply:
Done. Thanks.

Reply:
Done.